### ISSUE TITLE
Support for other editors

### ISSUE THREAD

**Issue by egeerardyn** [AlpenglowTheme/alpenglow-theme#24]
We should make AlpenGlow available for other editors such as
- [Atom](https://atom.io),
- [CudaText](http://uvviewsoft.com/cudatext/) as requested in https://github.com/YabataDesign/afterglow-theme/issues/127, and
- [VS Code](https://www.visualstudio.com/en-us/products/code-vs.aspx) as requested in https://github.com/YabataDesign/afterglow-theme/issues/128,
  - see also https://code.visualstudio.com/Docs/customization/themes
- [JetBrains IDEs]() (ItelliJ, CLion, ...) to mimic Afterglow support as provided by [AfterglowIntelliJ](https://github.com/JustSid/AfterglowIntelliJ) and [Jetbrains Afterglow Scheme](https://github.com/boneskull/jetbrains-afterglow-scheme)

Perhaps it also makes sense to see what other editors require (e.g. I can imagine that Electron-based editors (such as Atom and VS Code)) might just need minor tweaks between them to function properly).

**Comment by clstudio** [comment]
Great idea

**Comment by kaicataldo** [comment]
Hey there, I love Alpenglow Monokai and already created a port (using Atom's built in converter - I was expecting to have to do it by manually but the auto convert worked really well!): https://atom.io/themes/alpenglow-monokai-atom-syntax

Would the idea be to create other repos for other editors such as Atom under the AlpenglowTheme org (i.e. AlpenglowTheme/alpenglow-theme-atom, etc.)? I'd be happy to help with converting the other syntax themes over to Atom and would be open to discussing transferring ownership of the repo for my published theme to the org (and potentially helping to maintain the Atom themes, if you're open to having the help!).

Let me know what you'd like to do - I can get started converting over and publishing the Atom themes immediately once we hammer out the details of where they should live and have a repo set up for them so I start making PRs! :)

**Comment by kaicataldo** [comment]
As an aside, has there been any discussion about maintaining a fork of https://github.com/YabataDesign/afterglow-itermcolors as well? I'd love to see a Vim theme someday :)

**Comment by egeerardyn** [comment]
@kaicataldo We are certainly open to having all things Alpenglow united under one organization and anybody who is willing to help.
The idea is indeed to create a separate repo within the organization per themee (Atom, ST, iTerm2, ...).

Without a doubt, we would be very happy to have you responsible for the Atom side of things under our banner (or even as a co-owner for the whole organization if you want).

There has not been any discussion about iTerm2 / vim, but if there is any demand for a certain theme, we are open to having those centralized in this organization.

**Comment by kaicataldo** [comment]
@egeerardyn Sounds good! If you want to add me as a co-owner, I'm happy to take on adding the repo and working on the themes right away :) I'll look into how adding owners to Atom's package manager works so that others in the org can publish too!

**Comment by kaicataldo** [comment]
Sorry - things got a bit busier and I haven't gotten to it yet. Will try to get to it in the next few days!

**Comment by kaicataldo** [comment]
Started working on this tonight. My initial thought was to use a monorepo approach and have all three themes (Alpenglow, Alpenglow Monokai, & Alpenglow Twilight) exist in one repo with the ability to publish each individually. This unfortunately couldn't be done because apm (Atom's package manager) uses git tags (it creates pre-publish commits with tags automatically on publish). As a result, I decided to take the approach of just creating an individual repo for each theme. Let me know if you can think of/know a better way!

I looked at trying to create a single theme that has options to allow the user to switch between the variations, but all the information I saw suggested this was a hack (seems hard to maintain) and that Atom has a different mechanism for loading stylesheets than it does JavaScript files, and so the UI themes wouldn't have access to the syntax stylesheets if I loaded them up programmatically (a lot of Atom's UI themes will automatically change colors to match the syntax theme, and this would not be possible then).

Published the Twilight theme and moved the Monokai theme under the ownership of the org. Will republish so it hopefully shows the AlpenglowTheme GitHub avatar in the Atom's package manager :) ~~Will work on the main Alpenglow theme next (though I created a placeholder repo already).~~ Actually had time to go ahead and do it! All three themes are now published.

I plan to update the README's with more of the info in the Sublime Theme's version and add screenshots to the two that don't have them.

Let me know if you want me to do anything different - excited to help share the Alpenglow love :)

**Comment by kaicataldo** [comment]
And sorry for all the small releases - kept thinking I was done for the night and then would come back to it 😅

**Comment by kaicataldo** [comment]
I've also created a port of [Alpenglow for Hyperterm](https://github.com/AlpenglowTheme/alpenglow-hyperterm)

**Comment by clawoflight** [comment]
I would also really enjoy a vim theme. specifically, a gui vim theme since modern terminal emulators support true colors (set termguicolors in .vimrc).
Could someone maybe make me a diff between monokai/molokai and alpenglow-monokai? Then I could contribute a vim gui color scheme.

**Comment by kaicataldo** [comment]
Here are the colors: https://github.com/AlpenglowTheme/alpenglow-hyper/blob/master/index.js#L68

**Comment by clawoflight** [comment]
Are that really all though? GUI supports much more colors than terminal.

And even with this, it would take a while to find the correct mappings,
unless the same json object exists for plain monokai?

On 04/09/2018 02:50 AM, Kai Cataldo wrote:
> Here are the colors: https://github.com/AlpenglowTheme/alpenglow-hyper/blob/master/index.js#L68
> 

-- 
GPG fingerprint: 871F 1047 7DB3 DDED 5FC4 47B2 26C7 E577 EF96 7808

**Comment by kaicataldo** [comment]
Right, it will definitely involve some effort. That being said, Hyper is built on Electron and supports true colors, since it’s really just a browser window.

Here’s the Atom syntax theme that I generated from the original Sublime Text theme, if that’s helpful: https://github.com/AlpenglowTheme/alpenglow-monokai-atom-syntax

And here’s a Vim color scheme I made and use based on the Material color scheme, if that’s useful as a base: https://github.com/kaicataldo/material.vim
